### ISSUE TITLE
Support substring matching in mason test

### ISSUE THREAD

**Issue by ben-albrecht** [chapel-lang/chapel#15068]
#14037 allows passing multiple tests to `mason test` rather than running all tests within a mason package:

```sh
mason test test/test1.chpl test/test2.chpl
```

This feature request is to support substring matching similar to `cargo test`, allowing the previous test command to be run with the following `mason test` commands:

```sh
mason test test1 test2
# or
mason test test
# or
mason test t
```

**Comment by rajshekhar7** [comment]
@ben-albrecht I would like to work on this issue.

**Comment by ben-albrecht** [comment]
@rajshekhar7 - OK thanks!

**Comment by ben-albrecht** [comment]
Closed in #15087
